Subject: lookup_mnt is required for subfs
From: Jeff Mahoney <jeffm@suse.com>

lookup_mnt is required by subfs to actually perform locking and refcounting
on vfsmounts. Otherwise, it looks through vfsmounts lockless which is prone
to races

This reverts:
ChangeSet 1.1938.322.102 2004/11/07 20:07:11 arjan@infradead.org
  [PATCH] remove unused lookup_mnt export
  
  lookup_mnt() is entirely unused outside of deep vfs internals, remove the
  export since it's quite a low level interface anyway
  
  Signed-off-by: Arjan van de Ven <arjan@infradead.org>
  Signed-off-by: Andrew Morton <akpm@osdl.org>
  Signed-off-by: Linus Torvalds <torvalds@osdl.org>

diff -Nru a/fs/namespace.c b/fs/namespace.c
--- a/fs/namespace.c	2005-03-01 11:52:14 -05:00
+++ b/fs/namespace.c	2005-03-01 11:52:14 -05:00
@@ -106,6 +106,8 @@
 	return found;
 }
 
+EXPORT_SYMBOL(lookup_mnt);
+
 static inline int check_mnt(struct vfsmount *mnt)
 {
 	return mnt->mnt_namespace == current->namespace;
